.github/workflows/algolia/src/syncAlgoliaIndex.ts

```diff
@@ -0,0 +1,130 @@
+import fs from "fs";
+import path from "path";
+import matter from "gray-matter";
+import algoliasearch from "algoliasearch";
+
+type DocsFrontMatter = {
+  content: string;
+  data: {
+    objectId: string;
+    title?: string;
+    description?: string;
+    slug?: string;
+    tags?: Array<string>;
+  };
+};
+
+const getAllFiles = (
+  dirPath,
+  arrayOfFiles: Array<{ inputPath: string; outputPath: string; type: string }>,
```

Comment:
If I understand correctly, we pass `arrayOfFiles`, push some data to this array and return the original reference. In general, mutating arguments is a risky practice because it may lead to bugs that are hard to investigate.

Because the second argument is `[]` in both function calls, how about removing it? To avoid confusion between the concept of a file and the concept of info on the file, we can slightly rename the function too. Looking at `getAllFiles` (just the name), I may expect it to return an array of file names, Buffers, streams and so on. How about:

```tsx
const getFileInfos = (pathToDir: string, type: "glossary" | "docs"): FileInfo[] => {
  const fileInfos: FileInfo[] = [];
  
  // ...
    fileInfos.push(...getFileInfos(path.resolve(pathToDir, fileOrDirName), type))
  // ...
  
  return fileInfos
}

?

Reply:
Hey Alex, this change will also require functionality testing again, so will keep it out of this PR.

Reply:
Also the function is recursive, so it might be good to have the second parameter to have that differentiation.

Reply:
The function has been renamed meanwhile.